### PR TITLE
I need help

### PR DESCRIPTION
**Pull request by SerhiiUnhurian**
[DEMO LINK](https://SerhiiUnhurian.github.io/react_carousel/)

### CODE REVIEW

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]
your demo link isn't working

**Comment by SerhiiUnhurian** [comment]
Screenshot 2023-11-23 at 11 26 52

**Comment by SerhiiUnhurian** [comment]
i cant make this deploy

**Comment by SerhiiUnhurian** [comment]
check it out now

**Review by lerastarynets — CHANGES_REQUESTED** [review]
Unfortunately the project via the demo link isn't working
Screenshot at Nov 23 12-17-24

**Comment by SerhiiUnhurian** [comment]
> Unfortunately the project via the demo link isn't working Screenshot at Nov 23 12-17-24

yes i know
i need help with it

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
The item width doesn't work, feel free in the fe_chat if you need a help
image

File: src/App.scss [hunk 1]
```diff
@@ -1 +1,12 @@
 // style go here
```
Comment (OleziO) [inline src/App.scss:1]: Remove all comments from the code.

File: src/components/Carousel.scss [hunk 2]
```diff
@@ -1 +1,57 @@
...
+
+  // &__wrapper {
+  //   width: 260px;
+  //   overflow: hidden;
+  // }
+
+  // &__wrapper {
+  //   // display: inline-block;
+  //   // transition: transform 0.3s ease-in-out;
+  // }
```
Comment (OleziO) [inline src/components/Carousel.scss:18]: Remove all comments from the code.

**Review by OleziO — CHANGES_REQUESTED** [review]
image
Carousel and inputs do not work

File: src/components/Carousel.scss (hunk 2 above)
Comment (SerhiiUnhurian) [reply src/components/Carousel.scss]: okay

File: src/App.tsx [hunk 3]
```diff
@@ -20,17 +25,102 @@ class App extends React.Component<{}, State> {
...
         {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+        <h1 data-cy="title">Carousel with {images.length} images</h1>
+
+        <div>
+          <label htmlFor="itemWidth">Item Width: </label>
+          <input
+            type="number"
+            value={itemWidth}
+            onChange={(event) => this.handleInputChange(
```
Comment (anastasiiavorobiova) [inline src/App.tsx:69]:
```suggestion
            onChange={this.handleInputChange}
```

File: src/App.tsx [hunk 4]
```diff
@@ -20,17 +25,102 @@ class App extends React.Component<{}, State> {
...
     return (
       <div className="App">
         {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+        <h1 data-cy="title">Carousel with {images.length} images</h1>
+
+        <div>
+          <label htmlFor="itemWidth">Item Width: </label>
+          <input
+            type="number"
```
Comment (anastasiiavorobiova) [inline src/App.tsx:67]:
```suggestion
            type="number"
            name="itemWidth"
```

File: src/App.tsx [hunk 5]
```diff
@@ -20,17 +25,102 @@ class App extends React.Component<{}, State> {
...
+    event:
+    React.ChangeEvent<HTMLInputElement>, key: keyof State,
+  ) => {
+    const value = key === 'infinite'
+      ? event.target.checked
+      : Number(event.target.value);
+
+    this.setState((prevState) => ({
+      ...prevState,
+      [key]: Number.isNaN(value) ? 0 : value,
```
Comment (anastasiiavorobiova) [inline src/App.tsx:45]:
```suggestion
      [event.target.name]: Number.isNaN(value) ? 0 : value,
```

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

File: src/App.tsx (hunk 3 above)
Comment (SerhiiUnhurian) [reply src/App.tsx:69]: but how will it work?
don't we need this event?

File: src/App.tsx (hunk 4 above)
Comment (SerhiiUnhurian) [reply src/App.tsx:67]: i need to delete type?

File: src/App.tsx [hunk 6]
```diff
@@ -20,17 +25,108 @@ class App extends React.Component<{}, State> {
...
+    React.ChangeEvent<HTMLInputElement>, key: keyof State,
+  ) => {
+    const value = key === 'infinite'
+      ? event.target.checked
+      : Number(event.target.value);
+
+    this.setState((prevState) => ({
+      ...prevState,
+      [event.target.name]: Number.isNaN(value) ? 0 : value,
+      // [key]: Number.isNaN(value) ? 0 : value,
```
Comment (volodymyr-soltys97) [inline src/App.tsx:46]: Remove comments
```suggestion
```

File: src/App.tsx [hunk 7]
```diff
@@ -20,17 +25,108 @@ class App extends React.Component<{}, State> {
...
     return (
       <div className="App">
         {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+        <h1 data-cy="title">Carousel with {images.length} images</h1>
+
+        <div>
+          <label htmlFor="itemWidth">Item Width: </label>
+          <input
+            // type="number"
```
Comment (volodymyr-soltys97) [inline src/App.tsx:68]: The same as above
```suggestion
```

File: src/components/Carousel.tsx [hunk 8]
```diff
@@ -1,18 +1,96 @@
...
+        ? newIndex
+        : images.length - frameSize);
+    }
+
+    setTransitionDuration(animationDuration);
+  };
+
+  return (
+    <div className="Carousel" style={{ width: widthContainer }}>
+      {/* <h1 data-cy="title">Carousel</h1> */}
```
Comment (volodymyr-soltys97) [inline src/components/Carousel.tsx:55]:
```suggestion
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code
Try to place the buttons for switching on the same level
Also when I select infinite slider the toggle button doesn't activate, if you haven't implemented this logic then it's better to remove this checkbox
image

File: src/components/Carousel.tsx (hunk 8 above)
Comment (SerhiiUnhurian) [reply src/components/Carousel.tsx:55]: please help me to fix infinite scroll if we need it here

File: src/App.tsx (hunk 3 above)
Comment (TarasHoliuk) [reply src/App.tsx:69]: at least now you also pass `itemWidth` as a separate argument, so it wouldn't work like this.

but code like this `onChange={(event) => this.handleInputChange(event)}` doesn't make sense it's equal to `onChange={this.handleInputChange}`. 

Explanation: to onChange prop you pass the function that accepts `event` as a prop. In the first variant, you pass an additional wrapper function, which inside calls the original function.

P.S. You may get `itemWidth` from `event.target.name`

File: src/App.tsx [hunk 9]
```diff
@@ -20,17 +25,103 @@ class App extends React.Component<{}, State> {
...
+        </div>
 
-        <Carousel />
+        <Carousel
+          images={images}
+          step={step}
+          frameSize={frameSize}
+          itemWidth={itemWidth}
+          animationDuration={animationDuration}
+          infinite={false}
```
Comment (TarasHoliuk) [inline src/App.tsx:123]: there is still static `false` instead of dynamic variable
You haven't pushed/commited the updated code properly

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
It seems you haven't pushed required changes (and haven't re-deployed solution) - please, do it and re-request review again

**Review by loralevitska — CHANGES_REQUESTED** [review]
Almost done) Let's improve some bugs
- fix this bug while `infinite` checkbox checked 
image 
- add max and min values for each field to avoid such bugs
image
image

**Review by loralevitska — APPROVED** [review]
Almost perfect) 
`max` value still doesn't work, please fix it for improvement
image

**Pull request closed without merging**